File: Homework/quoc_khanh/lesson_1/136.SingleNumber.py
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        prev = 0
        for num in nums:
            prev = prev ^ num
        return prev

s = Solution()
print(s.singleNumber([2,2,1]) == 1)
print(s.singleNumber([4,1,2,1,2]) == 4)
print(s.singleNumber([1]) == 1)

File: Homework/Son/136-single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var singleNumber = function(nums) {
    let numSet = new Set();
    for (let num of nums) {
        if (!numSet.has(num)) numSet.add(num)
        else if (numSet.has(num)) numSet.delete(num)
    }
    return numSet.values().next().value;
};